### PR TITLE
Python: Convert string to boolean if the binding variable is Boolean

### PR DESCRIPTION
**Pull request by pritampan**
This is the equivalent of the [Legacy Python PR ](https://github.com/apache/iceberg/pull/6843).

### CODE REVIEW

**Review by puchengy — APPROVED** [review]

File: python/pyiceberg/expressions/literals.py [hunk 1]
```diff
@@ -562,6 +562,13 @@ def _(self, type_var: DecimalType) -> Literal[Decimal]:
                 f"Could not convert {self.value} into a {type_var}, scales differ {type_var.scale} <> {abs(dec.as_tuple().exponent)}"
             )
 
+    @to.register(BooleanType)
+    def _(self, type_var: BooleanType) -> Literal[bool]:
+        if self.value.upper() in ['TRUE', 'FALSE']:
+            return BooleanLiteral(True if self.value.upper() == 'TRUE' else False)
```
Comment (Fokko) [inline python/pyiceberg/expressions/literals.py:568]:
```suggestion
            return BooleanLiteral(self.value.upper() == 'TRUE')
```

File: python/pyiceberg/expressions/literals.py [hunk 2]
```diff
@@ -562,6 +562,13 @@ def _(self, type_var: DecimalType) -> Literal[Decimal]:
                 f"Could not convert {self.value} into a {type_var}, scales differ {type_var.scale} <> {abs(dec.as_tuple().exponent)}"
             )
 
+    @to.register(BooleanType)
+    def _(self, type_var: BooleanType) -> Literal[bool]:
+        if self.value.upper() in ['TRUE', 'FALSE']:
```
Comment (Fokko) [inline python/pyiceberg/expressions/literals.py:567]: Can we introduce a variable `value_upper` so we call `.upper()` just once?

**Review by Fokko — COMMENTED** [review]
Thanks for opening this PR, and I think this is a great addition. Could you rebase against the latest master? Today we had a failing master.

File: python/pyiceberg/expressions/literals.py (hunk 2 above)
Comment (pritampan) [reply python/pyiceberg/expressions/literals.py]: thanks for your suggestions, made the changes.

**Comment by pritampan** [comment]
> Thanks for opening this PR, and I think this is a great addition. Could you rebase against the latest master? Today we had a failing master.

Thanks for reviewing, I have rebased this branch and addressed the comments.

**Comment by Fokko** [comment]
Thanks @pritampan for opening this PR and @puchengy for the review! 👏🏻

**Pull request merged**